src/payload.rs

```diff
@@ -215,31 +258,54 @@ impl TryFrom<(&VersionedTransaction, RpcSendTransactionConfigWithBlockList)>
             RpcSendTransactionConfigWithBlockList,
         ),
     ) -> Result<Self, Self::Error> {
+        // Check encoding first to fail early if it's invalid
+        if let Some(encoding) = config_with_blocklist
+            .config
+            .as_ref()
+            .and_then(|c| c.encoding)
+        {
+            match encoding {
+                UiTransactionEncoding::Base58 | UiTransactionEncoding::Base64 => {}
+                _ => return Err(PayloadError::UnsupportedEncoding),
+            }
+        }
+
         let tx_bytes = bincode::serialize(transaction)?;
 
+        // Get max_retries safely
+        let max_retries = config_with_blocklist
+            .config
+            .as_ref()
+            .and_then(|c| c.max_retries.map(|r| r as u32));
+
+        // Get skip_preflight safely
+        let skip_preflight = config_with_blocklist
+            .config
+            .as_ref()
+            .map(|c| c.skip_preflight)
+            .unwrap_or(false);
+
+        // Get skip_sanitize safely
+        let skip_sanitize = config_with_blocklist
+            .config
+            .as_ref()
+            .map(|c| c.skip_sanitize)
+            .unwrap_or(false);
+
+        // Get blocklist_pdas safely
+        let blocklist_pdas = match &config_with_blocklist.blocklist_pdas {
+            Some(pdas) => pdas.clone(),
+            None => Vec::new(),
+        };
```

Comment:
```suggestion
        let blocklist_pdas = &config_with_blocklist.blocklist_pdas.unwrap_or_default();
```

unwrap or default should get us a empty vec in the case of None if let compiler know the end result is Vec<Pubkey>